### PR TITLE
feat: `GitHubProject.getInstance()`, `project.getProperties()`

### LINKED ISSUE #66: Add project `databaseId` and make project level data available
### Please avoid duplicates

- [X] I checked [all open feature requests](https://github.com/gr2m/github-project/issues?q=is%3Aissue+is%3Aopen+label%3Afeature) and none of them matched my problem.

### What’s missing?

Currently, project level data is fetched and stored locally in state. It is never exposed by the `github-project` API.

You can see that here: https://github.com/gr2m/github-project/blob/2cc6d4eb6ea2001cf7faa0717bb1e915debefc74/api/lib/get-state-with-project-items.js#L50

Additionally, add `databaseId` to the `projectNext` query.

### Why?

To prevent additional queries when this data is needed.

### Alternatives you tried

Manually fetching the data.

### Would you be interested in contributing the feature?

- [X] yes

### PR DESCRIPTION
**Pull request by tmelliottjr**
Adds `databaseId` and other project level properties to the `GitHubProject` object.

Closes #66

### CODE REVIEW

**Review by gr2m — CHANGES_REQUESTED** [review]
We need a separate method to get project data, using getters that return undefined until a certain internal state is not a good API. The new API should always behave the same to the consumer, no matter the internal state.

**Comment by tmelliottjr** [comment]
> We need a separate method to get project data, using getters that return undefined until a certain internal state is not a good API. The new API should always behave the same to the consumer, no matter the internal state.

You're right! I'll rework this. I had something more akin to that originally, but ultimately went with the 'simpler' option here.

**Comment by gr2m** [comment]
Start with the the README update to show the new API, once we agree on that, the rest will be easy.

**Comment by tmelliottjr** [comment]
> Start with the the README update to show the new API, once we agree on that, the rest will be easy.

Will do, my plan is to start this tomorrow, but I may not get to it until Monday.

**Comment by tmelliottjr** [comment]
@gr2m [Here](https://github.com/gr2m/github-project/blob/e1e103467dd015e2c97d90f0668e58fde39ea1c5/README.md#projectget) is the updated readme adding references for the potential new API. 

Another path we can consider is making this available during initialization. This might also lend itself to some issues we're trying to solve internally. For example, instantiating with an non-existent project number provides no immediate feedback:

```js
const project = new GitHubProject({
  token: "<token>",
  org: "<org>",
  number: 9999,
});
```

It's not until we attempt to list items that we realize an issue exists:

```js
await project.items.list()
```

```bash
file:///github-project/api/lib/get-state-with-project-items.js:33
    projectNext.fields.nodes
                ^

TypeError: Cannot read properties of null (reading 'fields')
    at getStateWithProjectItems (file:///github-project/api/lib/get-state-with-project-items.js:33:17)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async listItems (file:///github-project/api/items.list.js:13:26)
    at async file:///github-project/.test/scratchpad.js:12:13

Node.js v18.2.0
```

**Comment by gr2m** [comment]
I'd suggest to merge https://github.com/gr2m/github-project/pull/72 first, then create a 2.x branch, and update this PR head from `main` to `2.x`. I'd add the change to #72. 

If we merge it first then I fear it'd add a lot merge conflicts

> [Here](https://github.com/gr2m/github-project/blob/e1e103467dd015e2c97d90f0668e58fde39ea1c5/README.md#projectget) is the updated readme adding references for the potential new API.

looks good 👍🏼 

> Another path we can consider is making this available during initialization

We could offer a factory method (e.g. a static class method such as `Project.get(options)`) as an alternative which would internally instantiate the `project` instance and run `project.get()`.

We should definitely improve error messages either way. For a better debugging experience, we could call `project.get()` before using any other methods.

Removing the synchronous instantiation altogether will make it harder to test & mock.

**Comment by gr2m** [comment]
Should we go with `Project.get()` and `project.get()`? I fear that might be confusing, as `Project.get()` would resolve with a `project` instance, and `project.get()` would resolve with with an object of project properties.

Maybe `Project.getInstance()` and `project.getProperties()`?

**Comment by tmelliottjr** [comment]
> Should we go with `Project.get()` and `project.get()`? I fear that might be confusing, as `Project.get()` would resolve with a `project` instance, and `project.get()` would resolve with with an object of project properties.
> 
> Maybe `Project.getInstance()` and `project.getProperties()`?

Definitely agree, that `Project.get()` and `project.get()` would be too confusing. `Project.getInstance()` and `project.getProperties` are perfect! I have been referring to the latter as `data`, but properties seems clearer and wouldn't imply that it also included `items`.

**Comment by gr2m** [comment]
Let me know if you plan on working on it. If not I can try to work on it myself, but can't make promises when I'll find the time right now

**Comment by tmelliottjr** [comment]
> Let me know if you plan on working on it. If not I can try to work on it myself, but can't make promises when I'll find the time right now

Yes! I plan to, but I'm working through some higher priority items atm.

File: README.md [hunk 1]
```diff
@@ -57,6 +57,10 @@ const project = new GitHubProject({
   },
 });
 
+// get project data
+const projectData = await project.get();
+console.log(projectData.description)
```
Comment (tenshiAMD) [inline README.md:62]: @tmelliottjr I think we can remove this line

**Review by tenshiAMD — CHANGES_REQUESTED** [review]

File: README.md [hunk 2]
```diff
@@ -236,6 +253,14 @@ function (fieldOptionValue, userValue) {
   </tbody>
 </table>
 
+### `project.getProperties()`
+
+```js
+const projectData = await project.getProperties();
+```
+
+Returns project level data, such as `url`, `title`, `description` and `databaseId`
```
Comment (gr2m) [inline README.md:262]:
```suggestion
Returns project level data `url`, `title`, `description` and `databaseId`
```

**Review by gr2m — APPROVED** [review]

**Pull request merged**